astropy/coordinates/earth.py

```diff
@@ -395,6 +403,33 @@ def get_itrs(self, obstime=None):
                                      for the location of this object at the
                                      default ``obstime``.""")
 
+    def get_gcrs_posvel(self, obstime):
+        """
+        Calculate the GCRS position and velocity of this object at the
+        requested ``obstime``.
+
+        Parameters
+        ----------
+        obstime : `~astropy.time.Time`
+            The ``obstime`` to calculte the GCRS position/velocity at.
+
+        Returns
+        --------
+        obsgeoloc : `~astropy.units.Quantity`
+            The GCRS position of the object
+        obsgeovel : `~astropy.units.Quantity`
+            The GCRS velocity of the object
+        """
+        itrs = self.get_itrs(obstime)
+        geocentric_frame = GCRS(obstime=obstime)
+        # GCRS position
+        obsgeoloc = itrs.transform_to(geocentric_frame).cartesian.xyz.to(u.m)
+
+        vel_arr = np.cross(V_EARTH, np.rollaxis(obsgeoloc, 0, obsgeoloc.ndim))
+        vel_arr = np.rollaxis(vel_arr, -1, 0)
+        obsgeovel = u.Quantity(vel_arr)*u.m/u.s
```

Comment:
Since `vel_arr` is definitely not a `Quantity`, I'd do `u.Quantity(vel_arr, u.m/u.s, copy=False)` -- quite a bit faster.